docker/main/ngen/funcs.sh

```diff
@@ -212,4 +212,95 @@ run_secondary_mpi_ssh_worker_node()
+
+    if [ "${1:?No args given to tar_and_copy}" == "--dry-run" ]; then
+        local _DRY_RUN="true"
+        shift
+    fi
+
+    if [ "${1:?No contents directory given to tar_and_copy}" == "--compress" ]; then
+        local _TAR_EXTRA_ARGS="-z"
+        shift
+    fi
+
+    local _CONTENTS_DIR="${1:?No contents directory given to tar_and_copy}"
+    local _TAR_FILE="${2:?No archive file given to tar_and_copy}"
+    local _DEST_DIR="${3:?No copy destination directory given to tar_and_copy}"
+
+    if [ ! -e "${_CONTENTS_DIR}" ]; then
+        >&2 echo "$(print_date) Error: tar_and_copy contents directory '${_CONTENTS_DIR}' does not exist!"
+        exit 1
+    elif [ ! -d "${_CONTENTS_DIR}" ]; then
+        >&2 echo "$(print_date) Error: tar_and_copy contents directory '${_CONTENTS_DIR}' exists but is not a directory!"
+        exit 1
+    elif [ ! -e "${_DEST_DIR}" ]; then
+        >&2 echo "$(print_date) Error: tar_and_copy copy destination directory '${_DEST_DIR}' does not exist!"
+        exit 1
+    elif [ ! -e "${_DEST_DIR}" ]; then
+        >&2 echo "$(print_date) Error: tar_and_copy copy destination directory '${_DEST_DIR}' exist but is not a directory!"
+        exit 1
+    elif [ -e "${_TAR_FILE}" ]; then
+        >&2 echo "$(print_date) Error: tar_and_copy archive file '${_TAR_FILE}' already exists!"
+        exit 1
+    fi
+
+    if [ "${_DRY_RUN:-}" == "true" ]; then
+        return 0
+    fi
+
+    tar -c ${_TAR_EXTRA_ARGS:-} -f "${_DEST_DIR}/${_TAR_FILE}" -C "${_CONTENTS_DIR}" .
+    #cp -a "${_TAR_FILE}" "${_DEST_DIR}/."
+    #rm "${_TAR_FILE}"
+}
+
+gather_output() {
+    echo "$(print_date) Gather from remote worker host ${JOB_OUTPUT_WRITE_DIR:?Job temp output dir not defined} dirs"
+    for i in $(echo "${MPI_HOST_STRING}" | sed 's/,/ /g'); do
+        _HOST_NAME=$(echo "${i}" | awk -F: '{print $1}')
+        if [ "$(hostname)" == "${_HOST_NAME}" ]; then
+            continue
+        fi
+        scp -q -r ${_HOST_NAME}:${JOB_OUTPUT_WRITE_DIR}/ ${JOB_OUTPUT_WRITE_DIR}/. &
+    done
+    for p in $(jobs -p); do
+        wait ${p}
+        _R=$?
+        if [ ${_R} -ne 0 ]; then
+            echo "$(print_date) Error: remote copying of output exited with error ${_R}"
+            exit ${_R}
+        fi
+    done
+}
+
+move_output_to_dataset()
+{
+    # $1 output directory
+    # $2 dataset directory
+
+    if [ ! -d ${1:?No output directory given for copying to dataset} ]; then
+        >&2 echo "$(print_date) Error: cannot move output from non-directory path '${1}' to output dataset!"
+        exit 1
+    elif [ ! -d ${2:?No output dataset directory given for copying} ]; then
+        >&2 echo "$(print_date) Error: cannot move output to non-directory path '${1}' for output dataset!"
+        exit 1
+    fi
+
+    if [ $(ls ${1} | grep '.csv' | wc -l) -gt 0 ]; then
+        echo "$(print_date) Archiving and copying output CSVs to output dataset"
+        tar_and_copy ${1} job-${JOB_ID:?}-output.tar ${2}
```

Comment:
I think because the goal was to reduce practical running time by reducing transferred file count specifically.  The size of the data was not of tremendous importance.  Compression would slow things, both now and later.  In contrast, since the configs were effectively duplicated here, and not the primary content of the output dataset, it made more sense to compress that data.